datafusion/proto/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -1777,6 +1777,8 @@ fn round_trip_datatype() {
     }
 }
 
+// TODO file a ticket about handling deprecated dict_id attributes
+#[allow(deprecated)]
```

Comment:
I don't know what to do here:
1. Should we remove the `field_id` from DataFusion serialization code (as it is planned for removal eventually in DataFusion)?
2. Should we un-deprecate `field_id` in arrow-rs?
3. Something else?

@avantgardnerio, @thinkharderdev and @Dandandan , this test was added in
 - https://github.com/apache/datafusion/pull/8457

However,  @brancz's upstream PR in arrow-rs has deprecated `dict_id` on `Field` and all related APIs with the idea it will be eventually removed (and become a detail of the IPC transport rather than visible in the Schema)
- https://github.com/apache/arrow-rs/pull/6873 upstream in arrow-rs

@thinkharderdev mentioned https://github.com/apache/arrow-rs/pull/6873#issuecomment-2543132904

> If people are using the dictionary ID it's probably for something that should properly be done with kv metadata instead.

So one possibility could be to provide some more documentation / writeup as an upgrade guide for arrow explaining how to use field metadata instead ?

Please advise

Reply:
> If people are using the dictionary ID it's probably for something that should properly be done with kv metadata instead

I mostly meant that if you give people 32 bits of space someone, somewhere will use it as metadata if they don't need it for anything else. I don't really have any particular use case in mind. 

I don't have any objection to removing the field from datafusion serde. When we added that test we were still fighting to try and ensure unique dict IDs but we've long since moved on and found better ways around it.

Reply:
Or maybe on second thought we should keep it around until it is fully removed from upstream?

Reply:
Yeah, that sounds like a reasonable approach. I'll update the comments to match if no one else has a usecase

Reply:
Agreed. Also, https://github.com/apache/datafusion/pull/8457#issuecomment-1845770442 suggests it was never important to set the dict id directly, just that it wasn't sent correctly on the wire and already called out what to do once we remove the dict id. There were in fact a [number](https://github.com/apache/arrow-rs/pull/6399) of [dict](https://github.com/apache/arrow-rs/pull/6444) [fixes](https://github.com/apache/arrow-rs/pull/6711) that were necessary for us to correctly communicate the dicts correctly over the wire as well.